include/net/lwm2m_client_utils.h

```diff
@@ -31,6 +31,13 @@ extern "C" {
 #endif
 
 #if defined(CONFIG_LWM2M_CLIENT_UTILS_SECURITY_OBJ_SUPPORT)
+
+enum security_event_type {
+        SECURITY_EVT_OFFLINE_REQUEST,
+        SECURITY_EVT_ONLINE_REQUEST
+};
+
+typedef void (*security_cb_handler_t)(enum security_event_type evt);
```

Comment:
I think this callback needs to return `int` so it can tell the engine that request have failed.